### DISCUSSION TITLE
Strip the binary?

### DISCUSSION THREAD

**Discussion by borestad** [dimo414/bkt#34] [Ideas]
Hi!

Is the debug info really needed in production? The static musl binary goes from `5.4Mb` to `1.2Mb`. I've been using a stripped version myself and haven't found any issues yet. A lot of other binaries ships stripped, i.e 
- https://github.com/Schniz/fnm
- https://github.com/ouch-org/ouch

<img width="864" alt="image" src="https://user-images.githubusercontent.com/286455/236639472-ae669173-5698-48f0-916f-a295aa09dcdc.png">

Just some food for thought 🙂

**Comment by dimo414** [comment]
Hmm, I'm not sold on the value of this (and one of the first google results I found [agrees](https://stackoverflow.com/a/73628573/113632)). Sure smaller binaries are nice but missing debug symbols is frustrating so I'd want to see some meaningful benefit to stripping that justifies losing that data. Notably, I don't see any performance win by stripping (on my machine the stripped binary is actually slightly slower, but that could be noise).

It's [easy enough to do](https://doc.rust-lang.org/cargo/reference/profiles.html#strip) so you're welcome to send a PR if you'd like, but I'll probably let it bake for a little while and see if anyone else weighs in before merging/declining it. 

> I've been using a stripped version myself and haven't found any issues yet.

Well the primary difference would be less debugging data if you encounter a crash, so I wouldn't expect to find any _issues_ with a stripped vs. non-stripped binary in typical usage.

PS: you posted this under "[Show and tell](https://github.com/dimo414/bkt/discussions/categories/show-and-tell)", I moved it to the "Ideas" category since it's not a demo / usage discussion.

**Reply by miaLDN** [reply]
I'd prefer symbols intact please.  It's easy enough for a user to strip.
